src/metabase/shared/util/internal/time.cljs

```diff
@@ -382,7 +387,8 @@
     (let [[t value-type] (iso-8601->moment+type t)
           t              (truncate t unit)]
       (moment+type->iso-8601 [t value-type]))
-    (.startOf ^moment/Moment t (name unit))))
+    (let [t (->moment t)]
+      (.startOf ^moment/Moment t (name unit)))))
```

Comment:
I think it's more natural to hint when the name is introduced, not when it's used.
```suggestion
    (let [^moment/Moment t (->moment t)]
      (.startOf t (name unit)))))
```